v1/format/format.go

```diff
@@ -849,26 +956,81 @@ func (w *writer) writeFunctionCallPlain(terms []*ast.Term, comments []*ast.Comme
 		args[i] = t
 	}
 	loc := terms[0].Location
-	return w.writeIterable(args, loc, closingLoc(0, 0, '(', ')', loc), comments, w.listWriter())
+	var err error
+	comments, err = w.writeIterable(args, loc, closingLoc(0, 0, '(', ')', loc), comments, w.listWriter())
+	if err != nil {
+		panic(err)
+	}
+	return comments
 }
 
 func (w *writer) writeWith(with *ast.With, comments []*ast.Comment, indented bool) []*ast.Comment {
-	comments = w.insertComments(comments, with.Location)
+	var err error
+	comments, err = w.insertComments(comments, with.Location)
+	if err != nil {
+		panic(err)
+	}
 	if !indented {
 		w.write(" ")
 	}
 	w.write("with ")
-	comments = w.writeTerm(with.Target, comments)
+	comments, err = w.writeTerm(with.Target, comments)
+	if err != nil {
+		panic(err)
+	}
 	w.write(" as ")
-	return w.writeTerm(with.Value, comments)
+	comments, err = w.writeTerm(with.Value, comments)
+	if err != nil {
+		panic(err)
+	}
+	return comments
+}
+
+func (w *writer) writeTerm(term *ast.Term, comments []*ast.Comment) ([]*ast.Comment, error) {
+	currentComments := make([]*ast.Comment, len(comments))
+	copy(currentComments, comments)
+
+	var backup bytes.Buffer
+	backup.Write(w.buf.Bytes())
```

Comment:
Copying the full buffer could get expensive for large Rego files. Would it be possible to instead save the amount of bytes written to the buffer up to this point, and on error, we drop any bytes that has been written beyond that marker?

Reply:
Great idea! Updated to get the length and truncate instead.